src/manager/checkpoint/manager/topdown.rs

```diff
@@ -22,47 +93,135 @@ impl CheckpointManager for TopDownCheckpointManager {
+        validator: &Address,
     ) -> anyhow::Result<()> {
-        todo!()
+        let nonce = self
+            .child_client
+            .ipc_read_gateway_state(self.child_tipset().await?)
+            .await?
+            .applied_topdown_nonce;
+
+        let submission_tip_set = self.submission_tipset(epoch).await?;
+        let top_down_msgs = self
+            .parent_client
+            .ipc_get_topdown_msgs(
+                &self.child_subnet.id,
+                &self.parent.gateway_addr,
+                submission_tip_set,
+                nonce,
+            )
+            .await?;
+
+        log::debug!(
+            "nonce: {:} for submission tip set: {:} at epoch {:} of manager: {:}, size of top down messages: {:}",
+            nonce, submission_tip_set, epoch, self, top_down_msgs.len()
+        );
+
+        // we submit the topdown messages to the CHILD subnet.
+        let topdown_checkpoint = TopDownCheckpoint {
+            epoch,
+            top_down_msgs,
+        };
+        let message = MpoolPushMessage::new(
+            self.parent.gateway_addr,
+            *validator,
+            ipc_gateway::Method::SubmitTopDownCheckpoint as MethodNum,
+            cbor::serialize(&topdown_checkpoint, "topdown_checkpoint")?.to_vec(),
+        );
+        let mem_push_response = self
+            .child_client
+            .mpool_push_message(message)
+            .await
+            .map_err(|e| {
+                log::error!("error submitting checkpoint at epoch {epoch:} for manager: {self:}");
+                e
+            })?;
+
+        // wait for the checkpoint to be committed before moving on.
+        let message_cid = mem_push_response.cid()?;
+        log::debug!(
+            "checkpoint at epoch {:} for manager: {:} published with cid: {:?}, wait for execution",
+            epoch,
+            self,
+            message_cid,
+        );
+        self.child_client.state_wait_msg(message_cid).await?;
+        log::debug!(
+            "checkpoint at epoch {:} for manager: {:} published with cid: {:?}, executed",
+            epoch,
+            self,
+            message_cid,
+        );
+
+        Ok(())
     }
 
     async fn should_submit_in_epoch(
         &self,
-        _validator: &Address,
-        _epoch: ChainEpoch,
+        validator: &Address,
+        epoch: ChainEpoch,
     ) -> anyhow::Result<bool> {
-        todo!()
+        self.child_client
+            .ipc_validator_has_voted_topdown(&self.child_subnet.gateway_addr, epoch, validator)
+            .await
+            .map_err(|e| {
+                log::error!(
+                    "error checking if validator has voted for manager: {self:} due to {e:}"
+                );
+                e
+            })
     }
 
     async fn presubmission_check(&self) -> anyhow::Result<bool> {
-        Ok(true)
+        let state = self.sync_subnet_gateway_state().await?;
+        Ok(state.initialized)
     }
 }
+
+async fn obtain_checkpoint_period(
```

Comment:
`obtain` sounds a bit weird, what about `get`, or `pull`, in case you want to make more explicit that we are pulling it from the chain state.
```suggestion
async fn get_checkpoint_period(
```